### PR TITLE
Fix spaces in columns

### PR DESCRIPTION
**Pull request by Jesus89**
Related to https://github.com/CartoDB/carto-vl/issues/522

- Fix bug in date-wrapper regarding columns with spaces
- Fix bug in aggregation-query regarding columns with spaces
- Add unit tests
- Fix typo

### CODE REVIEW

**Comment by Jesus89** [comment]
Found also that there is another issue in the `psql.query` parser with the following content:

```sql
      SELECT *, _cdb_schema_name(base_table)
        FROM CDB_Overviews(
            CDB_QueryTablesText($windshaft$
        WITH 
_cdb_res AS (
    SELECT
    GREATEST(1*CDB_XYZ_Resolution(CDB_ZoomFromScale(0)), 0.00014579206139598132)::double precision AS res,
    ST_MakeEnvelope(0,0,0,0) AS bbox,
    (2*2.220446049250313e-16::double precision) AS eps
),
_cdb_params AS (
    SELECT
      res,
      bbox,
      CEIL((ST_XMIN(bbox) - eps*ABS(ST_XMIN(bbox)))/res)*res AS xmin,
      FLOOR((ST_XMAX(bbox) + eps*ABS(ST_XMAX(bbox)))/res)*res AS xmax,
      CEIL((ST_YMIN(bbox) - eps*ABS(ST_YMIN(bbox)))/res)*res AS ymin,
      FLOOR((ST_YMAX(bbox) + eps*ABS(ST_YMAX(bbox)))/res)*res AS ymax
      FROM _cdb_res
)

        SELECT
            MIN(_cdb_query.cartodb_id)  AS cartodb_id,
            ST_SetSRID(
                ST_MakePoint(
                    AVG(ST_X(_cdb_query.the_geom_webmercator)),
                    AVG(ST_Y(_cdb_query.the_geom_webmercator))
                ), 3857
            ) AS the_geom_webmercator
            , best price AS best price
            , count(*) AS _cdb_feature_count
        FROM (
        SELECT
            "best price","cartodb_id","the_geom","the_geom_webmercator"
        FROM
            (SELECT "best price","cartodb_id","the_geom","the_geom_webmercator" FROM (SELECT *, 99 AS "best price" FROM all_day) as _cdb_query_wrapper ) _cdb_epoch_transformation ) _cdb_query, _cdb_params
        WHERE 
  (_cdb_query.the_geom_webmercator && _cdb_params.bbox) AND
    ST_X(_cdb_query.the_geom_webmercator) >= _cdb_params.xmin AND
    ST_X(_cdb_query.the_geom_webmercator) < _cdb_params.xmax AND
    ST_Y(_cdb_query.the_geom_webmercator) >= _cdb_params.ymin AND
    ST_Y(_cdb_query.the_geom_webmercator) < _cdb_params.ymax

        GROUP BY
            Floor(ST_X(_cdb_query.the_geom_webmercator)/_cdb_params.res),
            Floor(ST_Y(_cdb_query.the_geom_webmercator)/_cdb_params.res)
            , best price
        
    $windshaft$)
        );
```
**Parser error**
```
syntax error at or near "AS"
    at Connection.parseE (/home/ubuntu/www/node-windshaft/current/node_modules/pg/lib/connection.js:573:11)
    at Connection.parseMessage (/home/ubuntu/www/node-windshaft/current/node_modules/pg/lib/connection.js:393:17)
```

The same query but using `"best_prices"` instead of `"best prices"` does not launch the parser error.

```sql
        SELECT
            "best_price","cartodb_id","the_geom","the_geom_webmercator"
        FROM
            (SELECT "best_price","cartodb_id","the_geom","the_geom_webmercator" FROM (SELECT *, 99 AS "best_price" FROM all_day) as _cdb_query_wrapper ) _cdb_epoch_transformation 
[2018-08-16 13:03:13.649] [INFO] console - PSQL 
      SELECT *, _cdb_schema_name(base_table)
        FROM CDB_Overviews(
            CDB_QueryTablesText($windshaft$
        WITH 
_cdb_res AS (
    SELECT
    GREATEST(1*CDB_XYZ_Resolution(CDB_ZoomFromScale(0)), 0.00014579206139598132)::double precision AS res,
    ST_MakeEnvelope(0,0,0,0) AS bbox,
    (2*2.220446049250313e-16::double precision) AS eps
),
_cdb_params AS (
    SELECT
      res,
      bbox,
      CEIL((ST_XMIN(bbox) - eps*ABS(ST_XMIN(bbox)))/res)*res AS xmin,
      FLOOR((ST_XMAX(bbox) + eps*ABS(ST_XMAX(bbox)))/res)*res AS xmax,
      CEIL((ST_YMIN(bbox) - eps*ABS(ST_YMIN(bbox)))/res)*res AS ymin,
      FLOOR((ST_YMAX(bbox) + eps*ABS(ST_YMAX(bbox)))/res)*res AS ymax
      FROM _cdb_res
)

        SELECT
            MIN(_cdb_query.cartodb_id)  AS cartodb_id,
            ST_SetSRID(
                ST_MakePoint(
                    AVG(ST_X(_cdb_query.the_geom_webmercator)),
                    AVG(ST_Y(_cdb_query.the_geom_webmercator))
                ), 3857
            ) AS the_geom_webmercator
            , best_price AS best_price
            , count(*) AS _cdb_feature_count
        FROM (
        SELECT
            "best_price","cartodb_id","the_geom","the_geom_webmercator"
        FROM
            (SELECT "best_price","cartodb_id","the_geom","the_geom_webmercator" FROM (SELECT *, 99 AS "best_price" FROM all_day) as _cdb_query_wrapper ) _cdb_epoch_transformation ) _cdb_query, _cdb_params
        WHERE 
  (_cdb_query.the_geom_webmercator && _cdb_params.bbox) AND
    ST_X(_cdb_query.the_geom_webmercator) >= _cdb_params.xmin AND
    ST_X(_cdb_query.the_geom_webmercator) < _cdb_params.xmax AND
    ST_Y(_cdb_query.the_geom_webmercator) >= _cdb_params.ymin AND
    ST_Y(_cdb_query.the_geom_webmercator) < _cdb_params.ymax

        GROUP BY
            Floor(ST_X(_cdb_query.the_geom_webmercator)/_cdb_params.res),
            Floor(ST_Y(_cdb_query.the_geom_webmercator)/_cdb_params.res)
            , best_price
        
    $windshaft$)
        );
```

**Comment by Jesus89** [comment]
This is the final map config after the fix:

FROM

```json
{
    "buffersize": {
        "mvt": 1
    },
    "layers": [
        {
            "type": "mapnik",
            "options": {
                "sql": "SELECT \"best price\",\"cartodb_id\",\"the_geom\",\"the_geom_webmercator\" FROM (SELECT *, 99 AS \"best price\" FROM all_day) as _cdb_query_wrapper ",
                "aggregation": {
                    "columns": {},
                    "dimensions": {
                        "best price": "best price"
                    },
                    "placement": "centroid",
                    "resolution": 1,
                    "threshold": 1
                },
                "dates_as_numbers": true,
                "metadata": {
                    "geometryType": true,
                    "columnStats": {
                        "topCategories": 32768,
                        "includeNulls": true
                    },
                    "sample": {
                        "num_rows": 1000,
                        "include_columns": [
                            "best price",
                            "cartodb_id"
                        ]
                    }
                }
            }
        }
    ]
}
```

TO

```sql
        SELECT
            "best price","cartodb_id","the_geom","the_geom_webmercator"
        FROM
            (SELECT "best price","cartodb_id","the_geom","the_geom_webmercator" FROM (SELECT *, 99 AS "best price" FROM all_day) as _cdb_query_wrapper ) _cdb_epoch_transformation 
[2018-08-16 13:39:37.476] [INFO] console - PSQL 
      SELECT *, _cdb_schema_name(base_table)
        FROM CDB_Overviews(
            CDB_QueryTablesText($windshaft$
        WITH 
_cdb_res AS (
    SELECT
    GREATEST(1*CDB_XYZ_Resolution(CDB_ZoomFromScale(0)), 0.00014579206139598132)::double precision AS res,
    ST_MakeEnvelope(0,0,0,0) AS bbox,
    (2*2.220446049250313e-16::double precision) AS eps
),
_cdb_params AS (
    SELECT
      res,
      bbox,
      CEIL((ST_XMIN(bbox) - eps*ABS(ST_XMIN(bbox)))/res)*res AS xmin,
      FLOOR((ST_XMAX(bbox) + eps*ABS(ST_XMAX(bbox)))/res)*res AS xmax,
      CEIL((ST_YMIN(bbox) - eps*ABS(ST_YMIN(bbox)))/res)*res AS ymin,
      FLOOR((ST_YMAX(bbox) + eps*ABS(ST_YMAX(bbox)))/res)*res AS ymax
      FROM _cdb_res
)

        SELECT
            MIN(_cdb_query.cartodb_id)  AS cartodb_id,
            ST_SetSRID(
                ST_MakePoint(
                    AVG(ST_X(_cdb_query.the_geom_webmercator)),
                    AVG(ST_Y(_cdb_query.the_geom_webmercator))
                ), 3857
            ) AS the_geom_webmercator
            , "best price" AS "best price"
            , count(*) AS _cdb_feature_count
        FROM (
        SELECT
            "best price","cartodb_id","the_geom","the_geom_webmercator"
        FROM
            (SELECT "best price","cartodb_id","the_geom","the_geom_webmercator" FROM (SELECT *, 99 AS "best price" FROM all_day) as _cdb_query_wrapper ) _cdb_epoch_transformation ) _cdb_query, _cdb_params
        WHERE 
  (_cdb_query.the_geom_webmercator && _cdb_params.bbox) AND
    ST_X(_cdb_query.the_geom_webmercator) >= _cdb_params.xmin AND
    ST_X(_cdb_query.the_geom_webmercator) < _cdb_params.xmax AND
    ST_Y(_cdb_query.the_geom_webmercator) >= _cdb_params.ymin AND
    ST_Y(_cdb_query.the_geom_webmercator) < _cdb_params.ymax

        GROUP BY
            Floor(ST_X(_cdb_query.the_geom_webmercator)/_cdb_params.res),
            Floor(ST_Y(_cdb_query.the_geom_webmercator)/_cdb_params.res)
            , "best price"
        
    $windshaft$)
        );
```

File: lib/cartodb/backends/dataview.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ DataviewBackend.prototype.getDataview = function (mapConfigProvider, user, param
 
         var dataviewDefinition = getDataviewDefinition(mapConfig.obj(), dataviewName);
         if (!dataviewDefinition) {
-            const error = new Error(`Dataview '${dataviewName}' does not exists`);
+            const error = new Error(`Dataview '${dataviewName}' does not exist`);
```
Comment (jgoizueta) [inline lib/cartodb/backends/dataview.js]: ❤️

File: lib/cartodb/models/aggregation/aggregation-query.js [hunk 2]
```diff
@@ -117,17 +117,19 @@ const dimensionNames = (ctx, table) => {
...
+        return `"${dimension_name}"`;
+    }));
 };
 
 const dimensionDefs = ctx => {
     let dimensions = aggregateDimensions(ctx);
     return sep(Object.keys(dimensions).map(dimension_name => {
         const expression = dimensions[dimension_name];
-        return `${expression} AS ${dimension_name}`;
+        return `"${expression}" AS "${dimension_name}"`;
```
Comment (jgoizueta) [inline lib/cartodb/models/aggregation/aggregation-query.js]: Isn't `expression` something like `avg(value)` (which shouldn't be quoted)?

Comment (Jesus89) [reply lib/cartodb/models/aggregation/aggregation-query.js]: Mmm maybe "avg(value)" won't work, but the column must be quoted when there is no aggregation.

Comment (jgoizueta) [reply lib/cartodb/models/aggregation/aggregation-query.js]: Oops, I was mixing dimensions with aggregated columns, ignore my comment

**Comment by Jesus89** [comment]
Thanks!. I'm going to merge it.

**Pull request merged**